### PR TITLE
more refactor cdb2api

### PR DESCRIPTION
**Pull request by adizaimi**
* clear few unnecessary gotos
* don't free buffer that we know its null
* remove sirace under test/tools
* fix error code check

### CODE REVIEW

**Comment by riverszhang89** [comment]
Could you please fix the format.

File: cdb2api/cdb2api.c [hunk 1]
```diff
@@ -4067,12 +4040,15 @@ int cdb2_bind_param(cdb2_hndl_tp *hndl, const char *varname, int type,
 int cdb2_bind_index(cdb2_hndl_tp *hndl, int index, int type,
                     const void *varaddr, int length)
 {
-    int rc = 0;
     pthread_once(&init_once, do_init_once);
+    if (log_calls)
```
Comment (mohitkhullar) [inline cdb2api/cdb2api.c]: We should log it at end and print rc

File: cdb2api/cdb2api.c [hunk 2]
```diff
@@ -1976,38 +1975,32 @@ static int cdb2_effects_request(cdb2_hndl_tp *hndl)
...
-
     if (type ==
         RESPONSE_HEADER__SQL_RESPONSE) { /* This might be the error that
                                             happened within transaction. */
         hndl->firstresponse =
             cdb2__sqlresponse__unpack(NULL, len, hndl->first_buf);
-        hndl->error_in_trans =
-            cdb2_convert_error_code(hndl->firstresponse->error_code);
+        int error_code = hndl->firstresponse->error_code;
+        hndl->error_in_trans = cdb2_convert_error_code(error_code);
```
Comment (mohitkhullar) [inline cdb2api/cdb2api.c]: Why an extra variable?

Comment (adizaimi) [reply cdb2api/cdb2api.c]: left over from previous refactoring, will fix.

File: cdb2api/cdb2api.c (hunk 1 above)
Comment (adizaimi) [reply cdb2api/cdb2api.c]: ok, was trying to avoid the goto, but will do.
Thanks

Comment (adizaimi) [reply cdb2api/cdb2api.c]: hah, there is no rc for this one, there might be for the other one i removed.

Comment (adizaimi) [reply cdb2api/cdb2api.c]: I don't think there is any difference if we put at function entry.

Comment (mohitkhullar) [reply cdb2api/cdb2api.c]: rc was removed from this function, we usually log rc when the function gives non-zero returns too.

**Pull request merged**